tests/unittests/sources/test_smartos.py

```diff
@@ -605,162 +598,160 @@ def test_scripts_shebang_not_added(self):
             ["#!/usr/bin/perl", 'print("hi")', ""]
         )
 
-        dsrc = self._get_ds(mockdata=my_returns)
+        m_jmc_client_factory.return_value = PsuedoJoyentClient(my_returns)
+        dsrc = ds()
         ret = dsrc.get_data()
-        self.assertTrue(ret)
-        self.assertEqual(
-            my_returns["user-script"], dsrc.metadata["user-script"]
-        )
+        assert ret
+        assert my_returns["user-script"] == dsrc.metadata["user-script"]
 
-        legacy_script_f = "%s/user-script" % self.legacy_user_d
-        self.assertTrue(os.path.exists(legacy_script_f))
-        self.assertTrue(os.path.islink(legacy_script_f))
+        legacy_script_f = "%s/user-script" % legacy_user_d
+        assert os.path.exists(legacy_script_f)
+        assert os.path.islink(legacy_script_f)
         shebang = None
         with open(legacy_script_f, "r") as f:
             shebang = f.readlines()[0].strip()
-        self.assertEqual(shebang, "#!/usr/bin/perl")
+        assert shebang == "#!/usr/bin/perl"
 
-    def test_userdata_removed(self):
+    def test_userdata_removed(self, ds, legacy_user_d, m_jmc_client_factory):
         """
         User-data in the SmartOS world is supposed to be written to a file
         each and every boot. This tests to make sure that in the event the
         legacy user-data is removed, the existing user-data is backed-up
         and there is no /var/db/user-data left.
         """
 
-        user_data_f = "%s/mdata-user-data" % self.legacy_user_d
+        user_data_f = "%s/mdata-user-data" % legacy_user_d
         with open(user_data_f, "w") as f:
             f.write("PREVIOUS")
 
         my_returns = MOCK_RETURNS.copy()
         del my_returns["user-data"]
 
-        dsrc = self._get_ds(mockdata=my_returns)
+        m_jmc_client_factory.return_value = PsuedoJoyentClient(my_returns)
+        dsrc = ds()
         ret = dsrc.get_data()
-        self.assertTrue(ret)
-        self.assertFalse(dsrc.metadata.get("legacy-user-data"))
+        assert ret
+        assert not dsrc.metadata.get("legacy-user-data")
```

Comment:
Same question/suggestion on False values:
```suggestion
        assert dsrc.metadata.get("legacy-user-data") is False
```